tests/test_sft_trainer.py

```diff
@@ -444,10 +445,32 @@ def test_successful_lora_target_modules_default_from_main():
 
 ############################# Finetuning Tests #############################
 def test_run_causallm_ft_and_inference():
-    """Check if we can bootstrap and finetune tune causallm models"""
+    """Check if we can bootstrap and finetune causallm models"""
     with tempfile.TemporaryDirectory() as tempdir:
         _test_run_causallm_ft(TRAIN_ARGS, MODEL_ARGS, DATA_ARGS, tempdir)
-        _test_run_inference(tempdir=tempdir)
+        _test_run_inference(checkpoint_path=_get_checkpoint_path(tempdir))
+
+
+def test_run_causallm_ft_save_with_save_model_dir_save_strategy_no():
+    """Check if we can bootstrap and finetune causallm model with save_model_dir
+    and save_strategy=no. Verify no checkpoints created and can save model.
+    """
+    with tempfile.TemporaryDirectory() as tempdir:
+        save_model_args = copy.deepcopy(TRAIN_ARGS)
+        save_model_args.save_strategy = "no"
+        save_model_args.output_dir = tempdir
+
+        trainer = sft_trainer.train(MODEL_ARGS, DATA_ARGS, save_model_args, None)
+        logs_path = os.path.join(
+            tempdir, FileLoggingTrackerConfig.training_logs_filename
+        )
+        _validate_logfile(logs_path)
+        # validate that no checkpoints created
+        assert not any(x.startswith("checkpoint-") for x in os.listdir(tempdir))
+
+        sft_trainer.save(tempdir, trainer)
```

Comment:
This was the best way I could think of to test the `save()` without having to try to mock up the SFTTrainer. I had tried doing something like...
```py
    training_args = SFTConfig(**transformer_kwargs)
    trainer = SFTTrainer(
        model=MODEL_NAME,
        args=TRAIN_ARGS,
        train_dataset=TWITTER_COMPLAINTS_DATA,
    )
    with tempfile.TemporaryDirectory() as tempdir:
        sft_trainer.save(tempdir, trainer)
```

but this requires needing to preprocess the train args so it can be parsed into SFTConfig and requires data preprocessing to run otherwise hits error `AttributeError: 'str' object has no attribute 'column_names'` when trying to tokenize the dataset